app/models/user.rb

```diff
@@ -1,2 +1,101 @@
 class User < ApplicationRecord
+
+  has_many :ratings, foreign_key: "user_id", dependent: :destroy
+  has_many :comments, foreign_key: "user_id", dependent: :destroy
+  has_many :comment_likes, foreign_key: "user_id", dependent: :destroy
+
+  attr_accessor :remember_token, :activation_token, :reset_token
+
+  before_save :downcase_email
+  before_create :create_activation_digest
+
+  validates :name, presence: true, length: { maximum: Settings.maximum_length_name }
+  validates :email, presence: true, length: { maximum: Settings.maximum_length_email }, format: { with: Settings.valid_email_regex }, uniqueness: {case_sensitive: false}
+  validates :password, presence: true, length: { minimum: Settings.minimum_length_password }, allow_nil: true
+
+  has_secure_password
+
+  def check_type type
+    return self.type_of_user == type
+  end
+
+  def get_order_id
+    if Order.find_by(user_id: self.id, status: "pending")
+      return Order.find_by(user_id: self.id, status: "pending").id
+    else
+      return nil
+    end
+  end
+
+  def get_restaurant
+    return unless self.check_type "manager"
+    return Restaurant.where(manager_id: self.id).last
+  end
+
+  def activate
+    update_attributes(:activated => true)
```

Comment:
```suggestion
    update_attributes(activated: true)
```
=> là cấu trúc cũ nhé, hạn chế dùng